lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php

```diff
@@ -1946,6 +1946,16 @@ public function mapField(array $mapping): array
             throw MappingException::repositoryMethodCanNotBeCombinedWithSkipLimitAndSort($this->name, $mapping['fieldName']);
         }
 
+        if (isset($mapping['targetDocument']) && isset($mapping['discriminatorMap'])) {
+            trigger_deprecation(
+                'doctrine/mongodb-odm',
+                '2.2',
+                'Mapping both "targetDocument" and "discriminatorMap" on field "%s" in class "%s" is deprecated.',
```

Comment:
```suggestion
                'Mapping both "targetDocument" and "discriminatorMap" on field "%s" in class "%s" is deprecated. Only one of them can be used at a time',
```

I'd suggest to use one of them so the deprecation message is actionable